base_directory_file_download/models/ir_filesystem_file.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017-2018 Onestein (<http://www.onestein.eu>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import logging
+import os
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+from odoo.tools import human_size
+
+_logger = logging.getLogger(__name__)
+
+
+class IrFilesystemDirectoryLine(models.TransientModel):
+    _name = 'ir.filesystem.file'
+
+    name = fields.Char(required=True)
+    filename = fields.Char()
+    file_content = fields.Binary(compute='_compute_file')
+    stored_filename = fields.Char()
+    directory_id = fields.Many2one(
+        'ir.filesystem.directory',
+        string='Directory'
+    )
+
+    @api.multi
+    def _file_read(self, fname, bin_size=False):
+
+        def file_not_found(fname):
+            raise UserError(_(
+                '''Error while reading file %s.
+                Maybe it was removed or permission is changed.
+                Please refresh the list.''' % fname))
+
+        self.ensure_one()
+        r = ''
+        directory = self.directory_id.get_dir() or ''
```

Comment:
I think the 'or' here is unnecessary because in get_dir it always returns a string.